django/db/models/sql/compiler.py

```diff
@@ -938,26 +938,28 @@ def get_converters(self, expressions):
                     converters[i] = (convs, expression)
         return converters
 
-    def apply_converters(self, row, converters):
-        row = list(row)
-        for pos, (convs, expression) in converters.items():
-            value = row[pos]
-            for converter in convs:
-                value = converter(value, expression, self.connection)
-            row[pos] = value
-        return tuple(row)
+    def apply_converters(self, rows, converters):
```

Comment:
I understand it's to avoid an attribute lookup in the loop but `connection = self.connection` looks a bit unnatural. Maybe `apply_converters()` could instead be a staticmethod with `self.connection` passed as an argument? Does it affect performance?